multiqc/modules/atropos/atropos.py

```diff
@@ -0,0 +1,1356 @@
+            for key in (
+                'total_record_count',
+                'total_bp_counts')))
+        
+        if 'trim' in data:
+            self.atropos_trim_data[sample_id] = data['trim']
+            self.atropos_general_data[sample_id]['fraction_bp_trimmed'] = \
+                1.0 - data['trim']['formatters']['fraction_total_bp_written']
+            for mod_dict in data['trim']['modifiers'].values():
+                if 'fraction_records_with_adapters' in mod_dict:
+                    self.atropos_general_data[sample_id]['fraction_records_with_adapters'] = \
+                        mean(mod_dict['fraction_records_with_adapters'])
+                    break
+
+    def atropos_assets(self, parent):
+        parent.css['atropos-assets/css/multiqc_atropos.css'] = os.path.join(
+            os.path.dirname(__file__), 'assets', 'css', 'multiqc_atropos.css')
+        parent.js['atropos-assets/js/multiqc_atropos.js'] = os.path.join(
+            os.path.dirname(__file__), 'assets', 'js', 'multiqc_atropos.js')
+
+    def atropos_general(self, parent):
+        """Add some single-number stats to the basic statistics table at the
+        top of the report.
+        """
+        headers = OrderedDict()
+        headers['input_format'] = {
+            'title': 'Format',
+            'description': 'Input File Format'
+        }
+        headers['total_record_count'] = {
+            'title': 'M Seqs',
+            'description': 'Total Sequences (millions)',
+            'min': 0,
+            'scale': 'Blues',
+            'modify': lambda x: x / 1000000,
+            'shared_key': 'read_count'
+        }
+        headers['sum_total_bp_count'] = {
+            'title': 'M bp',
+            'description': 'Total Base Pairs (millions)',
+            'min': 0,
+            'scale': 'Blues',
+            'modify': lambda x: x / 1000000,
+            'shared_key': 'base_count'
+        }
+        headers['mean_sequence_length'] = {
+            'title': 'Length',
+            'description': 'Average Sequence Length (bp)',
+            'min': 0,
+            'suffix': 'bp',
+            'scale': 'RdYlGn',
+            'format': '{:.0f}'
+        }
+        headers['fraction_bp_trimmed'] = {
+            'title': '% Base Pairs Trimmed',
+            'description': '% Total Base Pairs trimmed',
+            'max': 100,
+            'min': 0,
+            'suffix': '%',
+            'scale': 'RdYlBu-rev',
+            'modify': lambda x: x * 100,
+            'format': '{:.1f}%'
+        }
+        headers['fraction_records_with_adapters'] = {
+            'title': '% Reads w/ Adapters',
+            'description': '% Total Reads with Adapters',
+            'max': 100,
+            'min': 0,
+            'suffix': '%',
+            'scale': 'RdYlBu-rev',
+            'modify': lambda x: x * 100,
+            'format': '{:.1f}%'
+        }
+        print('adding general stats', str(self.atropos_general_data), str(headers))
```

Comment:
Please remove this (presumably a debugging statement..?)

Reply:
done